tools/2d_auto_threshold/auto_threshold.xml

```diff
@@ -23,7 +23,9 @@
     '$th_method.method_id'
     '$th_method.block_size'
     '$th_method.threshold'
-    '$invert_output'
+    #if $invert_output:

```

Comment:
You can avoid that by simply using truevalue falsevalue for all booleans.

Reply:
I don't quite understand. We **have** `truevalue` and `falsevalue` set for `invert_output`:
```xml
<param name="invert_output" type="boolean" truevalue="True" falsevalue="False" ...
```

The issue is that `argparse` can't deal with boolean parameters like `True` or `False`:
https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse

And the only alternative to using switches like `--invert_output` seems to be to parse the boolean string by hand, which is rather error-prone (since `bool('False')` evaluates to `True`).